tools/rna_tools/pipmir/pipmir.xml

```diff
@@ -0,0 +1,230 @@
+
+    "$input_bam"
+
+    &&
+    samtools index test_sorted.bam test_sorted.bai
+
+    ## the tool requires the location of RNAfold binary
+    &&
+    RNAfold_location=\$(which RNAfold)
+
+    &&
+    PIPmiR PIPELINE
+
+    -a test_sorted.bam
+
+    ## genome source
+    #if $refGenomeSource.genomeSource == 'history':
+        -t "$refGenomeSource.ownFile"
+    #else
+        -t "$refGenomeSource.builtin.fields.path"
+    #end if
+
+    -o test
+
+    ## optional parameters
+    #if $params.settingsType == "custom":
+        ## default: 50
+        -l $params.min_precursor
+
+        ## default: 500
+        -L $params.max_precursor
+
+        ## default: 2
+        -s $params.step_size
+
+        ## default: 10
+        -m $params.min_read
+    #end if
+
+    ## default: 1
+    -p \${GALAXY_SLOTS:-1}
+
+    -R \$RNAfold_location
+
+]]>
+    </command>
+    <inputs>
+        <param name="input_bam" type="data"
+            format="sam,bam" label="Alignment"
+            help="The bam or sam file containing alignment of the read data."/>
+
+        <!-- Genome source. -->
+        <conditional name="refGenomeSource">
+            <param name="genomeSource" type="select"
+                label="Will you select a reference genome from your
+                history or use a built-in 2bit genome?"
+                help="The 2bit version of the genome against which
+                the reads were aligned.">
+                <option value="2bit" selected="True">
+                    Use a built-in 2bit genome</option>
+                <option value="history">
+                    Use a 2bit genome from my current history</option>
+            </param>
+            <when value="2bit">
+            <param name="builtin" type="select"
+                label="Select a reference genome">
+                <options from_data_table="2bit_genome">
+                    <filter type="sort_by" column="2"/>
+                    <validator type="no_options"
+                    message="No 2bit genomes are available for
+                    the selected input dataset"/>
+                </options>
+            </param>
+            </when>
+            <when value="history">
+                <param name="ownFile" type="data" format="twobit"
+                label="Select the reference genome" />
+            </when>
+        </conditional>
+
+        <!-- optional parameters -->
+        <conditional name="params">
+            <param name="settingsType" type="select"
+                label="Optional parameters"
+                help="You can use the default settings or
+                set custom values for any of pipmir's parameters.">
+              <option value="default">Use defaults</option>
+              <option value="custom">Full parameter list</option>
+            </param>
+            <when value="default" />
+            <!-- Full/advanced params. -->
+            <when value="custom">
+                <param name="min_precursor" type="integer"
```

Comment:
it was explicitly specified by the tool. but, it should be the usual length of plant miRNA precursor. ranging from ~50 to a couple of hundred.